common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAccount.java

```diff
@@ -57,20 +61,24 @@ public AzureActiveDirectoryAccount(IDToken idToken, String uid, final String uTi
      * @return
      */
     public static AzureActiveDirectoryAccount create(final IDToken idToken, ClientInfo clientInfo) {
+        final String methodName = "create";
+        Logger.entering(TAG, methodName, idToken, clientInfo);
 
         final String uid;
         final String uTid;
 
-        //TODO: objC code throws an exception when uid/utid is null.... something for us to consider
         if (clientInfo == null) {
-            uid = "";
-            uTid = "";
+            throw new IllegalArgumentException("ClientInfo cannot be null");
```

Comment:
nit: you can remove the else statement below

Reply:
Good catch. Resolved in 38ebc90009020ced6b8eae20e589e6e2bc9b0edf

The static constructors for the `Account` subclasses (in this commit) are awfully similar -- may refactor this out later as I think these methods provide only marginal value and require PowerMock to effectively isolate. Not going to be in this PR though.

Thanks @addev-ashish